lib/dao.js

```diff
@@ -130,6 +130,20 @@ function errorModelNotFound(idValue) {
   return error;
 }
 
+function determineOptions(Model, options, method, numArguments) {
+  var dataSource = Model.getDataSource();
+  var opts = dataSource.isTransaction && !options.transaction ? Object.assign(
+    options, {transaction: dataSource.transaction}) : options;
+  if (dataSource.connector[method].length < numArguments) {
+    if (opts.transaction) {
+      throw new Error(
+        g.f('The connector does not support {{method}} within a transaction', method));
```

Comment:
This will most likely crash the process on an unhandled exception. Can we report the error via the callback please?

It may be easier (and probably more robust too) to change this method to take care of method invocation to.

Here is an example to illustrate my idea:

```js
function invokeConnectorMethod(method, Model, args, options, cb) {
  const dataSource = Model.getDataSource();
  const numArguments = args.length + 3; 
  options = // add transaction if needed
  const optionsSupported = dataSource.connector[method].length >= numArguments;
  if (options.transaction && !optionsSupported) {
    return cb(new Error(/*...*/));
  }
  const connector = dataSource.connector;
  const fullArgs = [Model.modelName].concat(args);
  if (optionsSupported) fullArgs.push(options);
  fullArgs.push(cb);
  connector[method].apply(connector, fullArgs);
}
```

Thoughts?

Reply:
You're probably right! I did it this way because of your suggestion to throw an Exception : )
I will look into this now, but the reason why we probably can't do this `invokeConnectorMethod()` is because of irregular method signatures, see https://github.com/lehni/loopback-datasource-juggler/blob/e3d47d8ebb1b77e77ea638e10a512dfadb0aa066/lib/dao.js#L2473-L2475

But we could of course add this as an exception directly in `invokeConnectorMethod()`?

Reply:
Implemented in 7d41a8f19af2b034df3221f65e9f58e6dc6fb687